sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -323,3 +323,58 @@ class WhileStatementExec(
     body.reset()
   }
 }
+
+/**
+ * Executable node for RepeatStatement.
+ * @param condition Executable node for the condition.
+ * @param body Executable node for the body.
+ * @param session Spark session that SQL script is executed within.
+ */
+class RepeatStatementExec(
+  condition: SingleStatementExec,
+  body: CompoundBodyExec,
+  session: SparkSession) extends NonLeafStatementExec {
+
+  private object RepeatState extends Enumeration {
```

Comment:
Just a note, `treeIterator` is not _exactly_ the same because for `while` it iterates while condition is **true**, but for `repeat` it iterates while condition is **false**.

Reply:
this is a good point, but let's aim to solve this in a follow-up PR to simplify things a bit in this PR, since logic is correct.